### DISCUSSION TITLE
Subscriptions with arguments.

### DISCUSSION THREAD

**Discussion by tdonhauser** [gqty-dev/gqty#1578] [Q&A]
Hallo,

I need help setting up a subscription. I want so setup a subscription for the following GraphQL schema:

```
type DataObject {
  name: String
  value: Number
}

type Subscription {
  dataObject(targetName: String!): DataObject!
}
```

I have a react component that renders the subscription like this:

```
export const MyComponent= () => {
    const { dataObject } = useSubscription();

    return (
      <p>
        New Notification: <b>{dataObject({ targetName: "myData" }).value}</b>
      </p>
    );
}
```

Everything works fine so far, but the component doesnt rerender on new data. If I use subscriptions without arguments it works. What am I doing wrong? 

I would be glad if someone can give me a hint how to solve it.
Thanks!

**Comment by vicary** [comment]
Current version of subscription client is lacking maintenance, our upcoming new major in [#1544](https://github.com/gqty-dev/gqty/pull/1544) uses a new subscription client which should covers this issue. Please stay tuned!

**Reply by tdonhauser** [reply]
Thanks for the info!

**Reply by vicary** [reply]
We are preparing for the beta launch for V3, please feel free to try out the new subscriptions!
